S2VX.Game/Editor/CommandPanel/CommandPanelInputBar.cs

```diff
@@ -4,41 +4,37 @@
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Sprites;
 using osu.Framework.Graphics.UserInterface;
-using osuTK.Graphics;
 using S2VX.Game.Story.Command;
 using System;
 using System.Collections.Generic;
 
 namespace S2VX.Game.Editor.CommandPanel {
     public class CommandPanelInputBar : FillFlowContainer {
         public Dropdown<string> DropType { get; } = new BasicDropdown<string> { Width = 160 };
-        public TextBox TxtStartTime { get; } = CreateErrorTextBox();
-        public TextBox TxtEndTime { get; } = CreateErrorTextBox();
+        public CommandPanelValueInput StartTime { get; }
+        public CommandPanelValueInput EndTime { get; }
         public CommandPanelValueInput StartValue { get; } = new();
         public CommandPanelValueInput EndValue { get; } = new();
         public Dropdown<string> DropEasing { get; } = new BasicDropdown<string> { Width = S2VXCommandPanel.InputSize.X };
         public Button BtnSave { get; }
 
-        public static BasicTextBox CreateErrorTextBox() =>
-            new() {
-                Size = S2VXCommandPanel.InputSize,
-                BorderColour = Color4.Red,
-                Masking = true
-            };
-
-        public static CommandPanelInputBar CreateAddInputBar(Action<ValueChangedEvent<string>> handleTypeSelect, Action handleAddClick) =>
-            new(false, handleTypeSelect, handleAddClick);
+        public static CommandPanelInputBar CreateAddInputBar(Action<ValueChangedEvent<string>> handleTypeSelect, Action handleAddClick,
+            Func<double> currentTimeDelegate) =>
+            new(false, handleTypeSelect, handleAddClick, currentTimeDelegate);
 
-        public static CommandPanelInputBar CreateEditInputBar(Action handleSaveClick) =>
-            new(true, _ => { }, handleSaveClick);
+        public static CommandPanelInputBar CreateEditInputBar(Action handleSaveClick, Func<double> currentTimeDelegate) =>
+            new(true, _ => { }, handleSaveClick, currentTimeDelegate);
```

Comment:
Just noticed this, but it looks like `handleTypeSelect` does nothing right now for edit input bar. This might be pretty weird for someone trying to use this drop down in the edit input bar. Not sure if we need to really address this though.

Reply:
That's right, `HandleTypeSelect` will only be called from the `AddInputBar` because it needs to reload the command list to apply the new filter. No such action needs to be done when changing the type for the edit bar.
![image](https://user-images.githubusercontent.com/12549910/118380933-bb7c2b00-b59a-11eb-9578-e010604afe04.png)